src/main/java/dbaccessor/user/UserAccessor.java

```diff
@@ -33,20 +33,18 @@ public long insert(User user) {
         return id;
     }
 
-    /** Gets the UserID of the user having the given username and email-id */
-    public Optional<Long> getUserIdFromUsernameAndEmail(String username, String emailID) {
-        String SQLStatment = "SELECT UserID FROM User WHERE Username=@Username AND EmailID=@EmailID";
+    /** Gets the UserID of the user having the given email-id */
+    public OptionalLong getUserIdFromEmail(String emailID) {
+        String SQLStatment = "SELECT UserID FROM User WHERE EmailID=@EmailID";
         Statement statement = Statement.newBuilder(SQLStatment)
-                                .bind("Username")
-                                .to(username)
                                 .bind("EmailID")
                                 .to(emailID)
                                 .build();
-        List<User> resultSet = spannerTemplate.query(User.class, statement, new SpannerQueryOptions().setAllowPartialRead(true));
+        ImmutableList<User> resultSet = ImmutableList.copyOf(spannerTemplate.query(User.class, statement, new SpannerQueryOptions().setAllowPartialRead(true)));
```

Comment:
FYI: Since you are never exposing this list outside of this method, you don't really need to make an immutable copy of it.

Reply:
Changed.